File: backend/app/schemas/survey.py
from datetime import datetime
from typing import Any
from pydantic import BaseModel, Field


class QuestionBase(BaseModel):
    id: str
    type: str
    title: str
    description: str = ""
    required: bool = False
    order: int = 0
    config: dict = Field(default_factory=dict)
    validation: dict = Field(default_factory=dict)


class SurveyCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=200)
    description: str = ""
    settings: dict = Field(default_factory=dict)
    questions: list[QuestionBase] = Field(default_factory=list)


class SurveyUpdate(BaseModel):
    title: str | None = Field(None, min_length=1, max_length=200)
    description: str | None = None
    settings: dict | None = None
    questions: list[QuestionBase] | None = None


class SurveyStatusUpdate(BaseModel):
    status: str = Field(..., pattern="^(draft|published|closed)$")


class SurveyOut(BaseModel):
    id: str
    title: str
    description: str
    status: str
    settings: dict
    questions: list[dict]
    created_at: datetime
    updated_at: datetime
    published_at: datetime | None = None

    class Config:
        from_attributes = True


class SurveyListItem(BaseModel):
    id: str
    title: str
    description: str
    status: str
    question_count: int = 0
    response_count: int = 0
    created_at: datetime
    updated_at: datetime
    published_at: datetime | None = None

    class Config:
        from_attributes = True

File: frontend/src/types/survey.ts
export interface Question {
  id: string
  type: string
  title: string
  description: string
  required: boolean
  order: number
  config: Record<string, any>
  validation: Record<string, any>
}

export interface Survey {
  id: string
  title: string
  description: string
  status: 'draft' | 'published' | 'closed'
  settings: Record<string, any>
  questions: Question[]
  created_at: string
  updated_at: string
  published_at: string | null
}

export interface SurveyListItem {
  id: string
  title: string
  description: string
  status: string
  question_count: number
  response_count: number
  created_at: string
  updated_at: string
  published_at: string | null
}

export interface Pagination {
  page: number
  page_size: number
  total: number
  total_pages: number
}

export interface SurveyListResponse {
  code: number
  message: string
  data: SurveyListItem[]
  pagination: Pagination
}
